File: feature_tests/java/somelib/src/main/java/dev/diplomattest/somelib/Two.java
package dev.diplomattest.somelib;

import dev.diplomattest.somelib.ntv.*;


import java.lang.foreign.Arena;
import java.lang.foreign.MemorySegment;
import java.lang.foreign.SegmentAllocator;
import java.lang.ref.Cleaner;
import java.util.List;
import static java.lang.foreign.ValueLayout.*;
import java.nio.charset.StandardCharsets;
import java.util.stream.Stream;

public class Two {

    MemorySegment internal;
    Cleaner.Cleanable cleanable;

    List<Object> selfEdges = List.of();
    List<Object> aEdges = List.of();
    List<Object> bEdges = List.of();
    

    static class TwoCleaner implements Runnable {

        MemorySegment segment;
        TwoCleaner(MemorySegment segment) {
            this.segment = segment;
        }

        public void run() {
            somelib_h.Two_destroy(this.segment);
        }
    }

    Two() {}
    Two(MemorySegment handle, List<Object> selfEdges, List<Object> aEdges, List<Object> bEdges) {
        this.internal = handle;
        this.selfEdges = selfEdges;
        this.aEdges = aEdges;
        this.bEdges = bEdges;
        

    }
    
    
}

File: feature_tests/kotlin/somelib/src/main/kotlin/dev/diplomattest/somelib/Two.kt
package dev.diplomattest.somelib;
import com.sun.jna.Callback
import com.sun.jna.Library
import com.sun.jna.Native
import com.sun.jna.Pointer
import com.sun.jna.Structure

internal interface TwoLib: Library {
    fun Two_destroy(handle: Pointer)
}

class Two internal constructor (
    internal val handle: Pointer,
    // These ensure that anything that is borrowed is kept alive and not cleaned
    // up by the garbage collector.
    internal val selfEdges: List<Any>,
    internal val aEdges: List<Any?>,
    internal val bEdges: List<Any?>,
    internal var owned: Boolean,
)  {

    init {
        if (this.owned) {
            this.registerCleaner()
        }
    }

    private class TwoCleaner(val handle: Pointer, val lib: TwoLib) : Runnable {
        override fun run() {
            lib.Two_destroy(handle)
        }
    }
    private fun registerCleaner() {
        CLEANER.register(this, Two.TwoCleaner(handle, Two.lib));
    }

    companion object {
        internal val libClass: Class<TwoLib> = TwoLib::class.java
        internal val lib: TwoLib = Native.load("diplomat_feature_tests", libClass)
    }

}